node-fs-readfile/cat.js

```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-console */
+
+const fs = require('fs');
+
+let index = 2;
+
+const id = setInterval(() => {
+  if (index < process.argv.length) {
+    fs.readFile(process.argv[index], 'utf8', (err, data) => {
+      if (err) throw err;
+      console.log(data);
+      index += 1;
+    });
+  } else if (index === process.argv.length) {
+    clearInterval(id);
+  }
+}, 1000);
```

Comment:
this does get the job done, but in general, we try to avoid using timers to "wait" until something is done. this leads to race conditions or poor performance or both. in the case that 1 second is not long enough, the program will be incorrect sometimes. even if 1 second is long enough, we're talking about the difference between waiting multiple seconds vs a few milliseconds.